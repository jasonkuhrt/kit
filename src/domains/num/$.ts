/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
