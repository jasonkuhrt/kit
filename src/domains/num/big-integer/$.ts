export * as BigInteger from './$$.ts'
