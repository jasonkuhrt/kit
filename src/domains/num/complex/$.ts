export * as Complex from './$$.js'
