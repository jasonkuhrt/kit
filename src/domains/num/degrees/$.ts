// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
