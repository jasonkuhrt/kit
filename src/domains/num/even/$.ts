// Export the namespace containing all Even operations
export * as Even from './$$.ts'
