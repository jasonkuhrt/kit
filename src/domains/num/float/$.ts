// Export the namespace containing all Float operations
export * as Float from './$$.js'
