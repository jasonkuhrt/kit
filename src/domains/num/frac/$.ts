// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
