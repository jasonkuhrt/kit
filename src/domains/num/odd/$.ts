// Export the namespace containing all Odd operations
export * as Odd from './$$.js'
