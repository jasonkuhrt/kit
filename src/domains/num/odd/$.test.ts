import { Test } from '#test'
import { Odd } from './$.js'

Test.describe('is')
  .on(Odd.is)
  .casesInput(
    // Valid cases - odd integers
    [1],
    [3],
    [-1],
    [-3],
    [5],
    [99],
    // Invalid cases - even integers
    [0],
    [2],
    [4],
    [-2],
    [-4],
    [100],
    // Invalid cases - non-integers
    [1.5],
    [3.1],
    [Infinity],
    [NaN],
    ['3'],
    [null],
  )
  .test()

Test.describe('from')
  .on(Odd.from)
  .cases(
    // Valid cases
    [1],
    [3],
    [-1],
    [-3],
    [5],
    [99],
    // Invalid cases - even integers should throw
    [0],
    [2],
    [4],
    [-2],
    [-4],
    // Invalid cases - non-integers should throw
    [1.5],
    [3.1],
    [Infinity],
    [NaN],
  )
  .test()

Test.describe('tryFrom')
  .on(Odd.tryFrom)
  .cases(
    // Valid cases
    [1],
    [3],
    [-1],
    [-3],
    [5],
    [99],
    // Invalid cases - should return null
    [0],
    [2],
    [4],
    [-2],
    [-4],
    [1.5],
    [3.1],
    [Infinity],
    [NaN],
  )
  .test()
