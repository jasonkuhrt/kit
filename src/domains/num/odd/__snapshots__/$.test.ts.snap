// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`from > from(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-2
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be odd, got: -2
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-3) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-3
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-3
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-4) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-4
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be odd, got: -4
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be odd, got: 0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be an integer, got: 1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
2
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be odd, got: 2
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(3) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
3
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(3.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.1
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be an integer, got: 3.1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(4) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
4
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be odd, got: 4
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
5
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(99) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
99
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
99
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be an integer, got: Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be an integer, got: NaN
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is("3") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-2
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-3) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-3
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-4) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-4
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
2
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(3) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(3.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(4) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
4
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
5
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(99) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
99
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-2
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-3) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-3
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-3
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-4) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-4
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
2
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(3) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
3
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(3.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.1
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(4) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
4
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
5
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(99) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
99
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
99
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;
